standardhealth/shr-tools
# Update to FHIR 3.0.2 and 4.0.1

**WHEN MERGING, PLEASE USE SQUASH AND MERGE**

**NOTE: DESCRIPTION UPDATED NOV 25 TO REFLECT RECENT UPDATES**

Updates definition files to 3.0.2 and 4.0.1 and updates logic to reflect new version numbers where necessary.

Also includes a minor fix related to cleaning up extensions that are not used in profiles.

Lastly, updates the FHIR IG Publisher to: ~`FHIR IG Publisher Version 1.0.0-SNAPSHOT - Built 2019-11-19T20:12:38.173+11:00 - Git 0bc883e6329b`~ `FHIR IG Publisher Version 1.0.2-SNAPSHOT - Built 2019-11-21T18:57:28.632+11:00 - Git 16d4b6a0782f`

To test:

* ~I think you should be able to use `shr-spec` `master` or `dev6` branch (I used `dev6`)~
* Use `shr-spec` `fix_bad_vs_urls` branch (which is based on `dev6` w/ two important fixes)

R4:
* `node . --clean -c ig-mcode/ig-mcode-r4-config.json /path/to/shr-spec/spec`
* `yarn ig:publish`
* `yarn ig:qa` -- there will be some expected errors about codesystems.html and unresolvable value sets.  That's ok.
* `yarn ig:open` -- just look around and ensure nothing is obviously wrong

STU3:
* ~set `"includeLogicalModels": false,` in the `shr-spec/spec/ig-all/ig-all-stu3-config.json` file~ _(no longer necessary)_
* `node . --clean -c ig-all/ig-all-stu3-config.json /path/to/shr-spec/spec`
* ~there is a known bug with AllergyIntolerance, so you need to:~
  * ~search for `obf-AllergyIntolerance` in `out/fhir/guide/ig.json` and remove its entry~
  * ~search for `obf-AllergyIntolerance` in `out/fhir/guide/resources/ig.xml` and remove its entry~
  * _the above hacks aren't necessary if you use the `fix_bad_vs_urls` branch_
* `yarn ig:publish` -- then wait a LOOOOONG time.  Go get lunch.  Then wait longer still.
* `yarn ig:qa` -- there will be lots of errors, but these are expected because its trying to use R4 profiles and extensions in STU3.  This is not a new problem.
* `yarn ig:open` -- just look around and ensure nothing is totally wrong
